packages/core/src/provider/__tests__/base.test.ts

```diff
@@ -102,6 +103,20 @@ describe("Base Tests", () => {
     `);
   });
 
+  it("should throw error if connected account has different entry point address than the provider", async () => {
+    const account = {
+      chain: polygonMumbai,
+      entryPointAddress: "0xOTHER_ENTRY_POINT_ADDRESS",
+      rpcClient: providerMock.rpcClient,
+      getAddress: async () => "0xMOCK_ADDRESS",
+      getFactoryAddress: () => "0xMOCK_FACOTRY_ADDRESS",
+      getEntryPointAddress: () => "0xOTHER_ENTRY_POINT_ADDRESS",
+      getOwner: () => undefined,
```

Comment:
hmm this may fail with zod being able to check if they are actually addresses at runtime (i.e. hex values, not 0x + english). considering making dummy values?

Reply:
yea could use SimpleAccount here too and init it with a different address than the provider

Reply:
from a "tests as documentation" perspective using SimpleAccount is probably a better example of how things work versus this